### ISSUE TITLE
Unable to import SimpleClassifier from simple_learn.Classifiers

### ISSUE THREAD

**Issue by mavismonica** [skekre98/simple_learn#39]
**Describe the bug**
Installed simple-learn. Tried importing SimpleClassifier from simple_learn.classifiers for trying the first SimpleLearn program. 
Faced " No module named 'simple_learn.classifiers' " error.

**To Reproduce**
1. Run 'pip install simple-learn'
2. On the python console, Run 'from simple_learn.Classifiers import SimpleClassifier'

**Expected behavior**
Should have been able to import SimpleClassifier without any error message.

**Screenshots**
![image](https://user-images.githubusercontent.com/61350590/103674249-d56b8880-4f7e-11eb-88f8-33430d3db0f2.png)

Also, if you check the site packages/simple_learn folder, there is no Classifiers present.
![image](https://user-images.githubusercontent.com/61350590/103674629-44e17800-4f7f-11eb-970c-bc731341ea6a.png)

**Desktop (please complete the following information):**
 - OS: [Windows 10]
 - Browser: [e.g. chrome, safari]
 - Version: [e.g. 0.0.1]

**Additional context**

**Comment by skekre98** [comment]
Hi @mavismonica, thanks for bringing this to our attention. It seems to me that you may be on an earlier version.  Can you try running the following command to upgrade simple-learn to the latest version and see if you get the same issue:
```
$ pip install simple-learn --upgrade
```

I think it also might be beneficial to add version information to the package. I will create a separate issue for this.

**Comment by Turtle24** [comment]
I replicated the same behaviour. I believe I have the most up-to-date version of simple learn aswell.

```
Name: simple-learn
Version: 0.0.11
Summary: A python package to simplify data modeling.
Home-page: https://github.com/skekre98/simple_learn
Author: Sharvil Kekre
Author-email: sharvildev@gmail.com
License: UNKNOWN
```
Here's the same error as @mavismonica 
```
>>> from simple_learn.classifiers import SimpleClassifier
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ModuleNotFoundError: No module named 'simple_learn.classifiers'
```

Additional context
I also tried it in the python shell and got the following. I might of gone off on a tangent for nothing so apologies if that's the case. However, what is the "clf" line meant to call to retrieve the json?  Also the clf.fit returns None, is it because the "__str__ method" isn't returning the json? Again if I'm way off sorry. Just trying to learn and I'm finding this all rather interesting. I think this is a seperate issue aswell, as it's not a module error.

clf returns the object. 
```
>>> from sklearn.datasets import load_iris
>>> from simple_learn.classifiers import SimpleClassifier
>>> iris = load_iris()
>>> clf = SimpleClassifier()
>>> clf.fit(iris.data, iris.target)
>>> clf
<simple_learn.classifiers.simple_classifier.SimpleClassifier object at 0x000001C039DAB790>
>>> 
```
clf_fit stores None but that makes sense as the fit method doesn't return anything in the code, is it meant to pass itself to the __str__ method?
```
>>> from sklearn.datasets import load_iris
>>> from simple_learn.classifiers import SimpleClassifier
>>> iris = load_iris()
>>> clf = SimpleClassifier()                 
>>> clf_fit = clf.fit(iris.data, iris.target)
>>> clf_fit
>>> print(clf_fit) 
None
```

**Comment by skekre98** [comment]
@Turtle24 thanks for the additional info. I am able to reproduce the issue now:
```
Python 3.7.6 (default, Jan  8 2020, 13:42:34) 
[Clang 4.0.1 (tags/RELEASE_401/final)] :: Anaconda, Inc. on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from simple_learn.classifiers import SimpleClassifier
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ModuleNotFoundError: No module named 'simple_learn.classifiers'
```

I believe the issue is because the classifiers module has not been added as an import in the package.
https://github.com/skekre98/simple_learn/blob/ac83490bde0f730367206623c3d835b9d24e3379/simple_learn/__init__.py#L21-L24

I'll mark this as a bug. Out of curiosity, how were you able to import SimpleClassifier here:
```
>>> from sklearn.datasets import load_iris
>>> from simple_learn.classifiers import SimpleClassifier
>>> iris = load_iris()
>>> clf = SimpleClassifier()                 
>>> clf_fit = clf.fit(iris.data, iris.target)
>>> clf_fit
>>> print(clf_fit) 
None
```

**Comment by skekre98** [comment]
I have fixed the import issue here: #47. However there is still the issue of the `__str__` function not working correctly which can be handled in a separate PR as I believe it is a different issue. @Turtle24 would you mind opening a separate bug issue to elaborate on this for tracking purposes?

**Comment by Turtle24** [comment]
@skekre98 it's a pleasure. I ran the python terminal from my simple_learn forked project. 
```
PS C:\Users\Aidan\simple_learn> python
Python 3.8.6 (tags/v3.8.6:db45529, Sep 23 2020, 15:52:53) [MSC v.1927 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> from simple_learn.classifiers import SimpleClassifier
>>> clf = SimpleClassifier()
>>> print(clf)
{
    "Type": "Empty Model",
    "Training Duration": "Nones",
    "GridSearch Duration": "Nones",
    "Parameters": {},
    "Metrics": {}
}
```
Here's a better look at the issue, seems there's a TypeError causing trouble.
```
>>> from sklearn.datasets import load_iris
>>> from simple_learn.classifiers import SimpleClassifier
>>> iris = load_iris()
>>> clf = SimpleClassifier()
>>> clf
<simple_learn.classifiers.simple_classifier.SimpleClassifier object at 0x000002039C23B790>
>>> print(clf) 
{
    "Type": "Empty Model",
    "Training Duration": "Nones",  
    "GridSearch Duration": "Nones",
    "Parameters": {},
    "Metrics": {}
}
>>> clf.fit(iris.data, iris.target)
>>> print(clf) 
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\Aidan\simple_learn\simple_learn\classifiers\simple_classifier.py", line 85, in __str__     
    return json.dumps(attr, indent=4)
  File "C:\Users\Aidan\AppData\Local\Programs\Python\Python38\lib\json\__init__.py", line 234, in dumps     
    return cls(
  File "C:\Users\Aidan\AppData\Local\Programs\Python\Python38\lib\json\encoder.py", line 201, in encode     
    chunks = list(chunks)
  File "C:\Users\Aidan\AppData\Local\Programs\Python\Python38\lib\json\encoder.py", line 431, in _iterencode
    yield from _iterencode_dict(o, _current_indent_level)
  File "C:\Users\Aidan\AppData\Local\Programs\Python\Python38\lib\json\encoder.py", line 405, in _iterencode_dict
    yield from chunks
  File "C:\Users\Aidan\AppData\Local\Programs\Python\Python38\lib\json\encoder.py", line 405, in _iterencode_dict
    yield from chunks
  File "C:\Users\Aidan\AppData\Local\Programs\Python\Python38\lib\json\encoder.py", line 438, in _iterencode
    o = _default(o)
  File "C:\Users\Aidan\AppData\Local\Programs\Python\Python38\lib\json\encoder.py", line 179, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
TypeError: Object of type int32 is not JSON serializable
>>>
```
Here's the proposed fix for the TypeError. 

```
class npEncoder(json.JSONEncoder):
    def default(self, obj):
        if isinstance(obj, np.int32):
            return int(obj)
        return json.JSONEncoder.default(self, obj)
def __str__(self):
. . .
return json.dumps(attr, cls=npEncoder, indent=4)
```
Result:
```
>>> from sklearn.datasets import load_iris
>>> from simple_learn.classifiers import SimpleClassifier
>>> iris = load_iris()
>>> clf = SimpleClassifier()
>>> clf.fit(iris.data, iris.target)
>>> print(clf) 
{
    "Type": "KNeighborsClassifier",
    "Training Duration": "0.0s",
    "GridSearch Duration": "0.11002588272094727s",
    "Parameters": {
        "metric": "euclidean",
        "n_neighbors": 4,
        "weights": "uniform"
    },
    "Metrics": {
        "Training Accuracy": 0.9866666666666667,
        "Jaccard Score": 0.9245283018867925,
        "F1 Score": 0.96
>>> iris = load_iris()
>>> clf = SimpleClassifier()
>>> clf.fit(iris.data, iris.target)
>>> print(clf)
{
    "Type": "KNeighborsClassifier",
    "Training Duration": "0.0s",
    "GridSearch Duration": "0.11002588272094727s",
    "Parameters": {
        "metric": "euclidean",
        "n_neighbors": 4,
        "weights": "uniform"
    },
    "Metrics": {
        "Training Accuracy": 0.9866666666666667,
        "Jaccard Score": 0.9245283018867925,
        "F1 Score": 0.96
    }
}
```
